### PR TITLE
Remove smoke test functionality to simplify infra

### PR DESCRIPTION
**Pull request by janeyx99**
Follow up to https://github.com/pytorch/pytorch/issues/73030

### CODE REVIEW

File: .github/scripts/generate_ci_workflows.py [hunk 1]
```diff
@@ -313,7 +309,7 @@ def _gen_test_jobs(self) -> Any:
             for shard in range(1, self.num_test_shards + 1):
                 test_jobs.append(
                     {
-                        "id": f"test_default_{shard}_{config['num_shards']}",
+                        "id": f"test_default_{shard}_{self.num_test_shards}",
```
[the remark this answers is not in the document]
Comment (suo) [reply .github/scripts/generate_ci_workflows.py:312]: oops. Well it just affects the name of the workflow in the yaml, not anything meaningful to the CI, so it probably just got passed through.

File: .github/scripts/generate_ci_workflows.py [hunk 2]
```diff
@@ -396,20 +392,6 @@ def generate_workflow_file(self, workflow_template: jinja2.Template) -> None:
             labels={LABEL_CIFLOW_DEFAULT, LABEL_CIFLOW_CPU, LABEL_CIFLOW_WIN}
         ),
     ),
-    CIWorkflow(
```
Comment (suo) [inline .github/scripts/generate_ci_workflows.py:399]: We should also set the workflow below this one to be `LABEL_CIFLOW_DEFAULT`, right?

**Review by suo — APPROVED** [review]

File: .github/scripts/generate_ci_workflows.py (hunk 2 above)
Comment (janeyx99) [reply .github/scripts/generate_ci_workflows.py:399]: ah good catch thanks

**Pull request closed without merging**